core/src/test/java/io/el/concurrent/DefaultEventLoopGroupTest.java

```diff
@@ -0,0 +1,159 @@
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.when;
+
+import java.util.Iterator;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.Executor;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ThreadFactory;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicReferenceFieldUpdater;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+
+@SuppressWarnings("unchecked")
+public class DefaultEventLoopGroupTest {
+
+  private final Executor mockExecutor = mock(Executor.class);
+  private final Runnable mockTask = mock(Runnable.class);
+  private final SingleThreadEventLoop eventLoop = new MockEventLoop(
+      new ThreadPerTaskExecutor(Executors.defaultThreadFactory())
+  );
+
+  private final EventLoopChooserFactory.EventLoopChooser chooser = () -> eventLoop;
+
+  private final EventLoopChooserFactory eventLoopChooserFactory = loops -> chooser;
+  private final DefaultEventLoopGroup defaultEventLoopGroup = new DefaultEventLoopGroup(10,
+      mockExecutor, eventLoopChooserFactory) {
+    @Override
+    protected EventLoop newChild(Executor executor) {
+      return eventLoop;
+    }
+
+    @Override
+    protected ThreadFactory newDefaultThreadFactory() {
+      return null;
+    }
+  };
+  private final EventLoopChooserFactory mockEventLoopChooserFactory = mock(
+      EventLoopChooserFactory.class);
+
+  @BeforeEach
+  public void setup() throws Exception {
+    when(mockEventLoopChooserFactory.newChooser(any())).thenReturn(chooser);
+  }
+
+  @Test
+  @DisplayName("When given mock chooser, then return random next event loop")
+  void testNext() {
+    EventLoop eventLoop = defaultEventLoopGroup.next();
+    assertNotNull(eventLoop);
+  }
+
+  @Test
+  @DisplayName("When call iterator, then return iterator")
+  void testIterator() {
+    Iterator<EventLoop> eventLoopIterator = defaultEventLoopGroup.iterator();
+    assertNotNull(eventLoopIterator);
+  }
+
+  @Test
+  @DisplayName("When shutdownGracefully, then isShuttingDown true")
+  void testShutdownGracefully() {
+    boolean isShuttingDown = defaultEventLoopGroup.shutdownGracefully(15, TimeUnit.SECONDS);
+    assertTrue(isShuttingDown);
+    assertTrue(defaultEventLoopGroup.next().isShuttingDown());
+  }
+
+  @Test
+  @DisplayName("When mock task executed and awaitTermination, then return true")
+  void testAwaitTermination() throws InterruptedException {
+    boolean awaitTermination = false;
+    try {
+      defaultEventLoopGroup.execute(mockTask);
+      awaitTermination = defaultEventLoopGroup.awaitTermination(1, TimeUnit.SECONDS);
+    } finally {
+      assertTrue(awaitTermination);
+      assertTrue(defaultEventLoopGroup.next().isTerminated());
+    }
+  }
```

Comment:
I think these simple test case is not need. I think it's better to test constructor